modules/uac_registrant/registrant.c

```diff
@@ -518,6 +520,13 @@ int run_reg_tm_cback(void *e_data, void *data, void *r_data)
 				contact = contact->next;
 			}
 		}
+		// if there is no expires in contact, try parse expires from header
+		if (exp == 0 && msg->expires) {
+			header_exp = (exp_body_t*)msg->expires->parsed;
```

Comment:
Even if the `Expires` hdr was (if exists) found (by parse_headers + EOH), the payload/body of the hdr is not automatically parsed, so `int parse_expires(struct hdr_field* _h)` should be called here.